.github/workflows/release_github.yaml

```diff
@@ -0,0 +1,113 @@
+name: Create a github release
+
+env:
+  BRANCH: ${{ github.ref_name }}
+  SNAP_VERSION: 3.3.1
+  VERSION: 0.0.1
+
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+      - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+
+jobs:
+  checks:
+    name: Check build condition for snap
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version:
+          - '3.9'
+          - '3.10'
+    steps:
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.BRANCH }}
+          fetch-depth: 0
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: ${{matrix.python-version}}
+          architecture: x64
+      - id: code-checks
+        name: Make sure the code follows conventions and tests correctly
+        run: |
+          make checks
+
+  autorelease:
+    name: Release the package on github
+    needs: checks
+    runs-on: 'ubuntu-latest'
+    strategy:
+      fail-fast: true
+    steps:
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.BRANCH }}
+          fetch-depth: 0
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: '3.10'
+          architecture: x64
+      - id: build-package
+        name: Build package
+        run: |
+          make checks
+          poetry build
+      - name: Add version to environment vars
+        run: |
+          echo "PROJECT_VERSION=${{ env.VERSION }}" >> $GITHUB_ENV
+      - name: Check if tag version matches project version
+        run: |
+          TAG=$(git describe HEAD --tags --abbrev=0)
+          echo $TAG
+          echo $PROJECT_VERSION
+          if [[ "$TAG" != "v$PROJECT_VERSION" ]]; then exit 1; fi
+#      - name: Release Notes
+#        uses: heinrichreimer/github-changelog-generator-action@v2.3
+#        with:
+#          token: ${{ secrets.SPARK_PKG_PUBLISH }}
+#          output: ".github/RELEASE-TEMPLATE.md"
+      - name: Create Github Release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.SPARK_PKG_PUBLISH }}
```

Comment:
Could be one liner https://cli.github.com/manual/gh_release_create
after signing https://docs.google.com/document/d/1OTV6l8LY0uBe4LehpP-Nd3QWB-2qXdxUJ3DYqka3w04/edit#